src/third_party/wiredtiger/src/include/btree_cmp.i

```diff
@@ -41,8 +61,8 @@ __wt_lex_compare(const WT_ITEM *user_item, const WT_ITEM *tree_item)
 	treep = tree_item->data;
 
 #ifdef HAVE_X86INTRIN_H
-	/* Use vector instructions if we'll execute at least 2 of them. */
-	if (len >= WT_VECTOR_SIZE * 2) {
+	/* Use vector instructions if we'll execute at least 1 of them. */
```

Comment:
@markbenvenuto 
To resolve build errors you can try adding CCFLAGS="-march=armv8-a+crc -flax-vector-conversions" along with scons build command.

On when to use vectorized code in _wt_lex_compare() routine we instrumented code and found that on an average Vector length = min (user-item-length, tree-item-length) is mostly greater than 16 bytes and less than 32 bytes for YCSB-Core-workloads (a, b, c, d, e & f) and for load-only (a & e) operations Vector length was greater than 32 bytes.

With above findings its more beneficial to use vectorized code for Vector length greater than 16 bytes.

Reply:
The reason the original value was chosen was because we tested and found that using vector instructions on the x86 was slower unless there were at least 2 vectors worth of bytes.

@prarkeri, how many bytes are required on the AArch64/Neon before the vector instructions are faster than the non-vector path?

We shouldn't make this change for the x86 unless the original analysis was wrong or the results have changed for expected customer hardware.

Reply:
@keithbostic 
Other than YCSB are there are any other benchmarks which target __wt_lex_compare() routine which we can use to better evaluate performance impact of proposed optimization?

Reply:
@prarkeri, I think the simplest approach would be a standalone program that calls `__wt_lex_compare()` using different byte lengths.

Reply:
@keithbostic 
Working on evaluating standalone benchmark for __wt_lex_compare() routine, will keep you posted on our findings shortly.

Reply:
@keithbostic 
We used synthetic benchmark (lets say wt_lex_comp_bench) which measures time spent in __wt_lex_compare() function for varying sizes of user and tree length. Below are our findings and recommendations:
* On aarch64 platform we found that in *general* vectorized code performance is better than C-implementation for vector size greater that 16 bytes.
* For x86_64 we will revert back to old changes where vectorized code will be used for vector size greater than 32 bytes. 

Let me know if above recommendation is fine, will push new changes?

Reply:
@prarkeri, that sounds great, thank you for testing!